src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2153,6 +2155,399 @@ impl Coordinator {
+            catalog.resolve_target_cluster(target_cluster, ctx.session()),
+            ctx
+        )
+        .id;
+        // TODO: Should this be passed to sequence_explain_analyze instead?
+        let otel_ctx = OpenTelemetryContext::obtain();
+        mz_ore::task::spawn(|| "sequence_explain_analyze", async move {
+            // EXPLAIN ANALYZE must be run as a single statement (see handle_execute where that is
+            // enforced). But we want all of the inner queries to be at the same timestamp with the
+            // correct timedomain, so explicitly start the session transaction.
+            return_if_err!(
+                ctx.session_mut().start_transaction(
+                    mz_ore::now::to_datetime(now),
+                    // Hard code these so that any session settings don't mess with them.
+                    Some(mz_sql::ast::TransactionAccessMode::ReadOnly),
+                    // TODO: Actually, should this be `None` so it picks up the current session var?
+                    // Need to think through what would happen to the various query results and if
+                    // that's what the user would want.
+                    Some(mz_sql::ast::TransactionIsolationLevel::StrictSerializable),
+                ),
+                ctx
+            );
+
+            let depends = match &plan.explainee {
+                Explainee::View(id)
+                | Explainee::MaterializedView(id)
+                | Explainee::Index(id)
+                | Explainee::ReplanView(id)
+                | Explainee::ReplanMaterializedView(id)
+                | Explainee::ReplanIndex(id) => btreeset! {*id},
+                Explainee::Statement(stmt) => stmt.depends_on(),
+            };
+            // TODO: Support all of them.
+            let (broken, plan, desc) = match plan.explainee {
+                plan::Explainee::Statement(plan::ExplaineeStatement::Select {
+                    broken,
+                    plan,
+                    desc,
+                }) => (broken, plan, desc),
+                _ => {
+                    ctx.retire(Err(AdapterError::Unsupported(
+                        "EXPLAIN ANALYZE with this explainee",
+                    )));
+                    return;
+                }
+            };
+            // GlobalIds stringify without risk of SQL injection.
+            let dependencies_in = depends.iter().map(|i| format!("'{i}'")).join(", ");
+
+            #[derive(Debug)]
+            enum AnalyzeQuery {
+                Sql(String),
+                Peek(PeekStage),
+            }
+
+            // Array of queries to execute and a function that receives the result rows.
+            let queries: [(
+                AnalyzeQuery,
+                Box<dyn Fn(&mut Vec<String>, Box<dyn RowIterator>) + Send>,
+            ); 4] = [
+                // Hitting persist w/o any filter pushdown. Use `EXPLAIN PLAN INSIGHTS` to find
+                // `sources` with no `op`.
```

Comment:
The other notices look super helpful, but I'm a bit concerned about this one, that it might be too noisy:
1. In many cases, a query is slow because the actual processing is big, and not because of reading the inputs without a pushdown. In such a case, pointing to filter pushdown would be a red herring.
2. Even if reading the inputs is indeed taking up a bi portion of the execution time, a significant filter pushdown might not be possible.

We could eventually sniff out 1. by looking at the elapsed CPU time of the source reading operators, and verifying that that is indeed what's taking a big portion of the execution time. But this is probably more complicated than the other things in this PR, so it's probably out of scope for this PR.